### ISSUE TITLE
mousedown on one view prevents mousedown on others

### ISSUE THREAD

**Issue by georeith** [paperjs/paper.js#993]
So I have a bunch of layered views. I have a script that detects mouse events on the upper view and proxies them (the literal HTML event not a paper event) in sequence to the views below as long as the events haven't had propagation stopped. Basically it lets me have multiple views but bind mouse events to them and have them propagate as if they were all on one view.

This was working fine until I recently updated and can narrow down the issue to `mousedown` events being denied on lower views because the upper view has `dragging` set.

The following code in [View.js](https://github.com/paperjs/paper.js/blob/develop/src/view/View.js#L1079-L1087) seems to be the cause:

```
	viewEvents[mousedown] = function(event) {
		var view = View._focused = getView(event);
		if (!dragging) {
			dragging = true;
			view._handleMouseEvent('mousedown', event);
		}
	};
```

What is the purpose of detecting dragging at the mousedown phase? Surely for a mousedown to occur the person can't be dragging (a mouseup must have occured). Or is this some strange multi-pointer workaround?

**Comment by lehni** [comment]
Yes, the purpose is to prevent multiple 'mousedown' events from firing when working with touch devices. Not sure there is a way to make both scenarios possible?

**Comment by lehni** [comment]
This was the issue in questoin: #922, and 52017340d2b005bc2afd7a0243aa17e3e755757c the fix

**Comment by georeith** [comment]
@lehni True although the current W3C recommendation which is [pointer events](https://www.w3.org/Submission/pointer-events/#dfn-primary-pointer) will fire multiple `pointerdown`. Perhaps a better way would be to align the events better for multi-touch, capture number of touch points, set the `isPrimary` boolean and provide that with the events.

I can see this being useful beyond my scenario, allowing someone to select multiple elements via touch .etc which just cancelling future mousedowns prevents.

**Comment by lehni** [comment]
Many things are useful, but we cannot cater for every possible use case out there. At the moment we don't support multi-touch pointer events, so it doesn't make sense to expect multiple mousedown events. Supporting multi-touch is a huge undertaking due to the lack of standardisation right now, so I think that's best covered by a seprate library that handles the mouse events (you don't have to use our mouse handlers, you can install your own and use `view.getEventPoint(event)`)

**Comment by lehni** [comment]
It is in a way similar to #992, both quite specific scenarios that are quite hard to cater for... I need to decide where to draw the line, it simply won't be able to do everything and make everybody happy.

**Comment by georeith** [comment]
@lehni True the real issue here is lack of support for rendering a single project across multiple canvases (for performance). I will probably revert to a single canvas for the time being and come back and see how that can be implemented into the library (this issue looks like it will crop up again) when there is more time.

**Comment by lehni** [comment]
It's also that I just reworked the whole event handling and it starts to feel like it's never going to be right... : )

**Comment by georeith** [comment]
@lehni Ha ha yeah I feel your pain. I did think it was bad time to bring up anything event related.

An alternate solution that would solve my case for is just being able to make direct calls to `view._handleMouseEvent`. Any chance we could expose that public (I know it technically is but calling `_` anything I don't own scares me).

Also should I be worried about `View._focused`? Is that just for keyboard events?
